x/genutil/types/types.go

```diff
@@ -0,0 +1,41 @@
+package types
```

Comment:
Why is this worthy of having to be under a specific directory?

Reply:
this file? or the `types` package as as a whole

Reply:
I think it is, these are type definitions here, might as well add them to remain consistent with our file organization